metals/src/main/scala/scala/meta/internal/metals/UserConfigurationOption.scala

```diff
@@ -15,4 +15,9 @@ case class UserConfigurationOption(
   def headerID: String = {
     title.toLowerCase().replace(' ', '-')
   }
+  def camelCaseKey: String =
+    key.split('-').toList match {
```

Comment:
You can actually pattern match on an Array:
```
  case Array(head, tail @_*) =>
    head ++ tail.flatMap(_.capitalize)
```
Not 100 % sure if more readable, but an interesting side note.

Reply:
Ahh, nice. I _think_ in terms of readability, I'd prefer the other one, but thanks!